components/Homepage/RoundActionCard/RoundActionCard.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+  Box,
+  Button,
+  Card,
+  CardBody,
+  Center,
+  HStack,
+  Heading,
+  Spinner,
+  Text,
+  VStack,
+} from "@chakra-ui/react";
+import { Phase } from "services/PhaseMgmtService";
+
+interface Props {
+  phase: Phase;
+  roundId: number;
+  isAuthed?: boolean;
+  hasSignedUp?: boolean;
+  hasVoted?: boolean;
+  hasSubmitted?: boolean;
+  onProfile: () => void;
+  onSignup: () => void;
+  onSignupAndJoinRound: () => void;
+  onJoinRound: () => void;
+  onVote: () => void;
+  onSubmit: () => void;
+  onRoundDetails: () => void;
+  loading?: boolean;
+}
+
+export const RoundActionCard = ({
+  phase,
+  roundId,
+  isAuthed,
+  hasSignedUp,
+  hasSubmitted,
+  hasVoted,
+  onProfile,
+  onSignup,
+  onSignupAndJoinRound,
+  onJoinRound,
+  onVote,
+  onSubmit,
+  onRoundDetails,
+  loading,
+}: Props) => {
+  const renderLabelContent = () => {
+    if (isAuthed) {
+      switch (phase) {
```

Comment:
nit - I'd write this as an object of type `{[key: Phase}: string}` so that you don't need the switch statement. I think the linter should technically complain since there's no default.